source/server/guarddog_impl.h

```diff
@@ -72,7 +72,7 @@ class GuardDogImpl : public GuardDog {
   /**
    * Exposed for testing purposes only (but harmless to call):
    */
-  int loopIntervalForTest() const { return loop_interval_.count(); }
+  int64_t loopIntervalForTest() const { return loop_interval_.count(); }
```

Comment:
Can you just have this return std::chrono::milliseconds or whatever the underlying type is?

Reply:
Since it is an internal, would an auto return type be acceptable?

Reply:
I don't know what the google style guide says on this, but I would prefer not.

Reply:
hmm, this is tricky, the type of the tick count is being figured as an `int64`

```
ERROR: C:/workspace/envoy/source/server/BUILD:98:1: C++ compilation of rule '//source/server:guarddog_lib' failed (Exit 2)
bazel-out/x64_windows-opt/bin/source/server/_virtual_includes/guarddog_lib\server/guarddog_impl.h(75): error C2440: 'return': cannot convert from '_Rep' to 'std::chrono::duration<__int64,std::milli>'
        with
        [
            _Rep=__int64
        ]
bazel-out/x64_windows-opt/bin/source/server/_virtual_includes/guarddog_lib\server/guarddog_impl.h(75): note: Constructor for class 'std::chrono::duration<__int64,std::milli>' is declared 'explicit'
Target //source/exe:envoy-static failed to build
```

Reply:
Don't return the count, just return the chrono object.

Reply:
sure, requires a test change then as well